src/modules/staking/staking.service.ts

```diff
@@ -56,6 +56,11 @@ export class StakingService {
     async init() {
         const chainId = this._signerService.chainId;
         this._stakingPoolFactoryAddress = chainConfigs()[chainId].stakingPoolFactoryAddress;
+
+        if (!this._stakingPoolFactoryAddress || !this._stakingPoolFactoryAddress.length) {
```

Comment:
Successful initialization of StakingService mean that it can be used, which will lead to errors when stakingPoolFactory isn't set. This should throw an error from my opinion. If we want to disable staking service, then it should be specified explicitly by additional parameter in init functions

Reply:
https://github.com/energywebfoundation/iam-client-lib/pull/370/files#diff-85024e9ac487a4d057efb7f1201aba9d16d5ee8d3c8c135a92c9005ce802a858R69

Then this change is incompatible - it's impossible to start any application on EWC right now as the staking address is empty and there is a forced connection to staking pool. IMO it makes sense to not initialize staking when staking pool factory address is empty.

https://github.com/energywebfoundation/iam-client-lib/blob/develop/src/init.ts#L57

Reply:
How about move this check in https://github.com/energywebfoundation/iam-client-lib/blob/5eaa4ec086f7fcdda92c1486688385d06293cd14/src/init.ts#L56 and return null stakingService if it can't be initialized? It allows to detect problems early